automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/PublishedWorkspaceSpec.scala

```diff
@@ -0,0 +1,130 @@
+              eventually {
+                isVisibleInLibrary(billingProject, workspaceName) shouldBe true
+              }
+            }
+
+            // should be able to be unpublished
+            Orchestration.library.unpublishWorkspace(billingProject, workspaceName)
+
+            // is not visible in the library table
+            // to keep the test from failing (let Elasticsearch catch up)
+            withClue("an unpublished workspace should not be visible in library.") {
+              eventually {
+                isVisibleInLibrary(billingProject, workspaceName) shouldBe false
+              }
+            }
+          }
+        }
+      }
+
+      "a cloned workspace" - {
+        "should be cloned without copying the published status" in {
+          val curatorUser = UserPool.chooseCurator
+          implicit val curatorAuthToken: AuthToken = curatorUser.makeAuthToken()
+
+          withCleanBillingProject(curatorUser) { billingProject =>
+            withWorkspace(billingProject, "PublishedWorkspaceSpec_workspace") { workspaceName =>
+
+              val data = LibraryData.metadataBasic + ("library:datasetName" -> workspaceName)
+              Orchestration.library.setLibraryAttributes(billingProject, workspaceName, data)
+              Orchestration.library.publishWorkspace(billingProject, workspaceName)
+
+              val clonedWorkspaceName = uuidWithPrefix("cloneWorkspace")
+
+              register cleanUp Orchestration.workspaces.delete(billingProject, clonedWorkspaceName)
+              Orchestration.workspaces.clone(billingProject, workspaceName, billingProject, clonedWorkspaceName)
+              // check cloned workspace does not have published status in workspace detail: "library:published": false
+              val response = Rawls.workspaces.getWorkspaceDetails(billingProject, clonedWorkspaceName)
+              response should include (""""library:published":false""")
+
+              // check cloned workspace is not visible in library
+              withClue("a cloned workspace should not be visible in library.") {
+                eventually {
+                  isVisibleInLibrary(billingProject, clonedWorkspaceName) shouldBe false
+                }
+              }
+
+            }
+          }
+        }
+
+        "should default to visible to 'all users' in Library Attributes" in {
+          val curatorUser = UserPool.chooseCurator
+          implicit val curatorAuthToken: AuthToken = curatorUser.makeAuthToken()
+
+          withCleanBillingProject(curatorUser) { billingProject =>
+            withWorkspace(billingProject, "PublishedWorkspaceSpec_workspace") { workspaceName =>
+
+              val data = LibraryData.metadataBasic + ("library:datasetName" -> workspaceName)
+              Orchestration.library.setLibraryAttributes(billingProject, workspaceName, data)
+              Orchestration.library.setDiscoverableGroups(billingProject, workspaceName, List("all_broad_users"))
```

Comment:
This needs to be set? By setting this doesn't it change the test to not be checking the "default" behavior?

Reply:
yes, it is needed. cloned workspace cannot inherit property `getDiscoverableGroup` from the workspace used in clone. that's verified at line 102.